pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/handler.go

```diff
@@ -0,0 +1,78 @@
+/*
+ Copyright 2021. The KubeVela Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package componentdefinition
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1alpha2"
+	"github.com/oam-dev/kubevela/pkg/oam/discoverymapper"
+	"github.com/oam-dev/kubevela/pkg/oam/util"
+)
+
+type handler struct {
+	client.Client
+	dm discoverymapper.DiscoveryMapper
+	cd *v1alpha2.ComponentDefinition
+}
+
+func (h *handler) CreateWorkloadDefinition(ctx context.Context) (util.WorkloadType, error) {
+	var workloadType = util.ComponentDef
+	var workloadName = h.cd.Name
+	if h.cd.Spec.Workload.Type != "" {
+		workloadType = util.ReferWorkload
+		workloadName = h.cd.Spec.Workload.Type
+	}
+	if h.cd.Spec.Schematic != nil && h.cd.Spec.Schematic.HELM != nil {
+		workloadType = util.HELMDef
+	}
+	wd := new(v1alpha2.WorkloadDefinition)
+	err := h.Get(ctx, client.ObjectKey{Namespace: h.cd.Namespace, Name: workloadName}, wd)
```

Comment:
if err !=nil{
   notfound 
}
...

Reply:
Got it, I will modify this part of the code logic